homeassistant/components/light/mipow.py

```diff
@@ -0,0 +1,140 @@
+"""
+Support for Mipow Bluetooth smartbulbs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.mipow/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICES, CONF_NAME
+from homeassistant.components.light import (
+    ATTR_RGB_COLOR, ATTR_WHITE_VALUE, SUPPORT_RGB_COLOR, SUPPORT_WHITE_VALUE,
+    Light, PLATFORM_SCHEMA)
+from homeassistant.util.color import color_rgb_to_rgbw, color_rgbw_to_rgb
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['https://github.com/mjg59/python-mipow/releases/download/0.3/'
+                'mipow-0.3.tar.gz#mipow==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_MIPOW_LED = (SUPPORT_RGB_COLOR | SUPPORT_WHITE_VALUE)
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA},
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a Mipow bulb."""
+    lights = []
+    for address, device_config in config[CONF_DEVICES].items():
+        device = {}
+        device['name'] = device_config[CONF_NAME]
+        device['address'] = address
+        light = MipowLight(device)
+        if light.is_valid:
+            lights.append(light)
+
+    add_devices(lights)
+
+
+class MipowLight(Light):
+    """Representation of a Mipow light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        # pylint: disable=import-error
+        import mipow
+
+        self._name = device['name']
+        self._address = device['address']
+        self._white = 0
+        self._rgb = (0, 0, 0)
+        self._state = False
+        self._bulb = mipow.mipow(self._address)
+        self._bulb.connect()
+        self.update()
+        self.is_valid = True
```

Comment:
I'm still not crazy about this pattern. Not sure where it started. I still think if there's an problem in __init__ we should be raising an exception and catching it in setup_platform. Otherwise we're keeping `is_valid` around for the lifetime of the object.